### ISSUE TITLE
Tokenizer for AMRBART-large-finetuned-AMR3.0-AMRParsing 

### ISSUE THREAD

**Issue by HenryCai11** [muyeby/AMRBART#13]
I noticed that for the finetuned AMRBarts, there are no tokenizers offered in the huggingface hub, whereas the v2 models have tokenizers with a different vocab size (v1 53844 vs. v2 53228). My questions are: 

1. Where can I get the tokenizers for those finetuned models?
2. Is there a demonstration for tokens used in v2 models (because I found that newly-added tokens used in v2 models are different from tokens illustrated in the paper)?
3. Is it OK for me to use BartTokenizer to load the pretrained AMR tokenizers? 

Thank you!

**Comment by goodbai-nlp** [comment]
Hi, 

Thanks for your interest!
- For the v1 model, we initialize the tokenizer from the vanilla BART tokenizer following SPRING where the additional tokens are added [here](https://github.com/goodbai-nlp/AMRBART/blob/acl2022/spring/spring_amr/tokenization_bart.py).  So there is no need to upload the tokenizer.
- For the v2 model. we re-implement the tokenizer [here](https://github.com/goodbai-nlp/AMRBART/blob/main/fine-tune/model_interface/tokenization_bart.py). You may initialize the tokenizer from the one we uploaded following [here](https://github.com/goodbai-nlp/AMRBART/blob/ffe79bdbe99cb51ef8313b4b206a862bd3f6f1d2/fine-tune/main.py#L131).
- The reason why v2 models have a small vocabulary size than v1 is that we use fewer special \<pointer:xx\> tokens and remove \<mask:xx\> tokens. You may compare the code [here](https://github.com/goodbai-nlp/AMRBART/blob/b5e45ae4d8b39e811b224c1ab60c87dd8b57cb61/spring/spring_amr/tokenization_bart.py#L68) and [here](https://github.com/goodbai-nlp/AMRBART/blob/ffe79bdbe99cb51ef8313b4b206a862bd3f6f1d2/fine-tune/common/additional-tokens.json#L2958).

Hope these comments can help you.

**Comment by HenryCai11** [comment]
Thank you so much!

**Comment by HenryCai11** [comment]
@goodbai-nlp Hi, sorry to bother again. I still wonder how should I initialize the toeknizer with AMRBartTokenizer.
```
from transformers import BartForConditionalGeneration, BartTokenizer, AutoConfig
from spring_amr.tokenization_bart import AMRBartTokenizer

config = AutoConfig.from_pretrained("xfbai/AMRBART-large-finetuned-AMR3.0-AMRParsing")
model = BartForConditionalGeneration.from_pretrained("xfbai/AMRBART-large-finetuned-AMR3.0-AMRParsing")
tokenizer = AMRBartTokenizer.from_pretrained("facebook/bart-large", config=config)
```
I tried initializing this way. However, length of the **tokenizer** did not match the vocab size in **config**. Did I miss the point for the initialization? Looking forward to your reply. Thank you!

**Comment by goodbai-nlp** [comment]
Hi, 

I assume you are trying to initialize the tokenizer in the v1 version. You may follow the code [here](https://github.com/goodbai-nlp/AMRBART/blob/b5e45ae4d8b39e811b224c1ab60c87dd8b57cb61/fine-tune/inference_amr.py#L59). Additionally, there is no need to pass the config parameter when initializing our tokenizer.

**Comment by HenryCai11** [comment]
Thank you!
